sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/InsertAdaptiveSparkPlan.scala

```diff
@@ -39,11 +41,26 @@ case class InsertAdaptiveSparkPlan(
 
   private val conf = adaptiveExecutionContext.session.sessionState.conf
 
+  def containShuffle(plan: SparkPlan): Boolean = {
+    plan.find {
+      case _: Exchange => true
+      case s: SparkPlan => !s.requiredChildDistribution.forall(_ == UnspecifiedDistribution)
+    }.isDefined
+  }
+
+  def containSubQuery(plan: SparkPlan): Boolean = {
+    plan.find(_.expressions.exists(_.find {
```

Comment:
It is absolutely ridiculous to do so here. We deliberately added `SubqueryAdaptiveNotSupportedException` to make sure that it's all-in or all-out for the main query and all its subqueries in order to ensure subquery reuse in AQE.

Reply:
It's still all-in or all-out. The complete check is `isSubquery || containShuffle(plan) || containSubQuery(plan)`, so we only call `containSubQuery` for the main query.